src/writers/writer.cpp

```diff
@@ -8,255 +8,451 @@
 
 namespace zarr = acquire::sink::zarr;
 
-/// DirectoryCreator
+namespace {
+/// Returns the index of the chunk in the lattice of chunks for the given frame
+/// and dimension.
+size_t
+chunk_lattice_index(size_t frame_id,
```

Comment:
Is `frame_id` still a good name here? Isn't this more like an index into the array in the given dimension (i.e. `dims[dimension_idx]`).

Reply:
It's using the frame id to compute and return that dimension.